test/CMakeLists.txt

```diff
@@ -0,0 +1,133 @@
+  string(REPLACE "." ";" MAJMIN "${vers}.X.Y")
+  list(GET MAJMIN 0 MAJ)
+  list(GET MAJMIN 1 MIN)
+  if (${MAJ} STREQUAL 0 )
+    if (NOT TARGET diffbinder)
+      add_executable( diffbinder diffbinder.cpp) 
+    endif()
+    add_test( NAME ${testname}_diff
+#--always-success option forces the diffbinder to return success regardles of the results of comparison. 
+      COMMAND diffbinder ${testnamenodot}.cpp ${testname}.ref  --always-success
+       WORKING_DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}")
+    SET_TESTS_PROPERTIES(${testname}_diff PROPERTIES DEPENDS  binder)
+    SET_TESTS_PROPERTIES(${testname}_diff PROPERTIES DEPENDS  ${testname}_foo )
+  else()
+    if (${MAJ} STREQUAL 2)
+      Python_add_library(${testnamenodot}${vers} MODULE  ${testname}.hpp  ${testnamenodot}.cpp)
+    endif()
+    if (${MAJ} STREQUAL 3)
+      Python3_add_library(${testnamenodot}${vers} MODULE  ${testname}.hpp  ${testnamenodot}.cpp)
+    endif()
+    set_source_files_properties( ${testnamenodot}.cpp PROPERTIES GENERATED TRUE)
+    add_dependencies(${testnamenodot}${vers}  ${testnamenodot}.cpp)
+    set_target_properties(${testnamenodot}${vers} PROPERTIES OUTPUT_NAME  ${testnamenodot}
+                                               NO_SYSTEM_FROM_IMPORTED ON
+                                               ARCHIVE_OUTPUT_DIRECTORY ${PROJECT_BINARY_DIR}/py${vers}/${CMAKE_INSTALL_LIBDIR}/$<0:> 
+                                               LIBRARY_OUTPUT_DIRECTORY ${PROJECT_BINARY_DIR}/py${vers}/${CMAKE_INSTALL_LIBDIR}/$<0:>
+                                               RUNTIME_OUTPUT_DIRECTORY ${PROJECT_BINARY_DIR}/py${vers}/${CMAKE_INSTALL_LIBDIR}/$<0:>)
+    target_compile_definitions(${testnamenodot}${vers} PRIVATE ${testnamenodot}_EXPORTS )
+    target_include_directories( ${testnamenodot}${vers} PRIVATE ${CMAKE_CURRENT_SOURCE_DIR}  ${pybind11_INCLUDE_DIR})
+
+    if(BINDER_USE_PYTHON_IN_TEST)
+      add_test( NAME ${testname}_python${vers} 
+      COMMAND ${Python_EXECUTABLE} -c "import ${testnamenodot}"
+        WORKING_DIRECTORY ${PROJECT_BINARY_DIR}/py${vers}/${CMAKE_INSTALL_LIBDIR}/$<0:>)
+    else()
+      add_test( NAME ${testname}_python${vers} 
+      COMMAND echo "Import for ${testnamenodot} is disabled"
+        WORKING_DIRECTORY ${PROJECT_BINARY_DIR}/py${vers}/${CMAKE_INSTALL_LIBDIR}/$<0:>)
+    endif()
+    SET_TESTS_PROPERTIES(${testname}_python${vers} PROPERTIES DEPENDS ${testnamenodot}${vers})
+  endif()
+endmacro( binder_test testname vers)
+set( binder_tests
+         T00.basic
+         T01.enum
+         T02.function
+         T05.default
+         T07.class
+         T08.constructor
+         T09.overload
+         T10.inheritance
+         T11.override
+         T12.operator
+         T15.copy
+         T15.inner_class
+         T17.anonymous
+         T20.template
+         T30.include
+         T31.include_for_class
+         T32.call_guard
+         T40.stl
+         T42.stl.names
+         T50.namespace_binder
+         )
```

Comment:
Yes. One can do something like "create tests for all files that match pattern".  Should be a line or two in cmake language.
But it would be better to have it in a separate MR.

Reply:
ok, deferring this to separate PR